baseplate/observers/tracing.py

```diff
@@ -216,7 +219,11 @@ def on_finish(self, exc_info: Optional[_ExcInfo]) -> None:
 
         self.end = current_epoch_microseconds()
         self.elapsed = self.end - typing.cast(int, self.start)
-        self.record()
+
+        for key, value in self.counters.items():
+            self.binary_annotations.append(self._create_binary_annotation(key, value))
```

Comment:
Any concerns about cardinality with this?  Also, should the `key` include something like `counter-` in it so we don't have to worry about some random metric colliding with something else?

Reply:
good points, will do

Reply:
Added a namespace. I don't think cardinality is an issue for tag values on a tracing system, and is no worse than `client.counter("...").increment()` for metrics. Do you think this API is too abusable?

Reply:
I don’t think so, just didn’t know if we needed to worry about tracing tag cardinality. 🙂